spyder/plugins/plots/widgets/figurebrowser.py

```diff
@@ -685,34 +696,74 @@ def save_figure_as(self, fig, fmt):
             save_figure_tofile(fig, fmt, fname)
 
     # ---- Thumbails Handlers
-    def add_thumbnail(self, fig, fmt):
-        thumbnail = FigureThumbnail(background_color=self.background_color)
-        thumbnail.canvas.load_figure(fig, fmt)
+    def _calculate_figure_canvas_width(self, thumbnail):
+        """
+        Calculate the witdh the thumbnail's figure canvas need to have for the
+        thumbnail to fit the scrollarea.
+        """
+        figure_canvas_width = (
+            self.scrollarea.width() -
+            2 * self.lineWidth() -
+            self.scrollarea.viewportMargins().left() -
+            self.scrollarea.viewportMargins().right() -
+            thumbnail.savefig_btn.width() -
+            thumbnail.layout().spacing()
+            )
+        if is_dark_interface():
+            # This is required to take into account some hard-coded padding
+            # and margin in qdarkstyle.
+            figure_canvas_width = figure_canvas_width - 6
```

Comment:
Is this looking ok on Win/Linux/MAc ( I can check mac :-p )

Reply:
I've only test on Windows unfortunately.

Reply:
Ok, I will check on linux and mac ;-)

Reply:
Thanks!